### ISSUE TITLE
Received signal 11 (Segmentation Fault)

### ISSUE THREAD

**Issue by MARMOTatZJU** [taichi-dev/taichi#1100]
I run Ubuntu 18.04 and Python3 within my Virtual Box machine (VBox works for GAMES-101 and pytorch).
I installed taichi from source: https://pypi.doubanio.com/simple.
pip3 reports that taichi has been installed successfully, but import taichi returns the following errors:

```
[E 06/01/20 22:32:27.711] Received signal 11 (Segmentation fault)


[38;2;255;000;255m***********************************
[0m[38;2;255;000;255m* Taichi Compiler Stack Traceback *
[0m[38;2;255;000;255m***********************************
[0m[38;2;255;000;255m/home/xuyinda/.local/lib/python3.6/site-packages/taichi/core/../lib/taichi_core.so: taichi::Logger::error(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool)
[0m[38;2;255;000;255m/home/xuyinda/.local/lib/python3.6/site-packages/taichi/core/../lib/taichi_core.so: taichi::signal_handler(int)
[0m[38;2;255;000;255m/lib/x86_64-linux-gnu/libc.so.6(+0x3ef20) [0x7f8dbbd28f20]
[0m[38;2;255;000;255m/usr/lib/x86_64-linux-gnu/libLLVM-9.so.1: llvm::StringMapImpl::LookupBucketFor(llvm::StringRef)
[0m[38;2;255;000;255m/usr/lib/x86_64-linux-gnu/libLLVM-9.so.1(+0x98ec5f) [0x7f8d97605c5f]
[0m[38;2;255;000;255m/usr/lib/x86_64-linux-gnu/libLLVM-9.so.1(+0x98ee0c) [0x7f8d97605e0c]
[0m[38;2;255;000;255m/usr/lib/x86_64-linux-gnu/libLLVM-9.so.1(+0x984fc5) [0x7f8d975fbfc5]
[0m[38;2;255;000;255m/usr/lib/x86_64-linux-gnu/libLLVM-9.so.1(+0x990e9d) [0x7f8d97607e9d]
[0m[38;2;255;000;255m/usr/lib/x86_64-linux-gnu/libLLVM-9.so.1: llvm::ManagedStaticBase::RegisterManagedStatic(void* (*)(), void (*)(void*)) const
[0m[38;2;255;000;255m/usr/lib/x86_64-linux-gnu/libLLVM-9.so.1: llvm::cl::OptionCategory::registerCategory()
[0m[38;2;255;000;255m/usr/lib/x86_64-linux-gnu/libLLVM-9.so.1(+0x8a1fa2) [0x7f8d97518fa2]
[0m[38;2;255;000;255m/lib64/ld-linux-x86-64.so.2(+0x10733) [0x7f8dbc0eb733]
[0m[38;2;255;000;255m/lib64/ld-linux-x86-64.so.2(+0x151ff) [0x7f8dbc0f01ff]
[0m[38;2;255;000;255m/lib/x86_64-linux-gnu/libc.so.6: _dl_catch_exception
[0m[38;2;255;000;255m/lib64/ld-linux-x86-64.so.2(+0x147ca) [0x7f8dbc0ef7ca]
[0m[38;2;255;000;255m/lib/x86_64-linux-gnu/libdl.so.2(+0xf96) [0x7f8dbb8c7f96]
[0m[38;2;255;000;255m/lib/x86_64-linux-gnu/libc.so.6: _dl_catch_exception
[0m[38;2;255;000;255m/lib/x86_64-linux-gnu/libc.so.6: _dl_catch_error
[0m[38;2;255;000;255m/lib/x86_64-linux-gnu/libdl.so.2(+0x1735) [0x7f8dbb8c8735]
[0m[38;2;255;000;255m/lib/x86_64-linux-gnu/libdl.so.2: dlopen
[0m[38;2;255;000;255m/usr/lib/x86_64-linux-gnu/libGLX_mesa.so.0(+0x55f51) [0x7f8d9d90ff51]
[0m[38;2;255;000;255m/usr/lib/x86_64-linux-gnu/libGLX_mesa.so.0(+0x43120) [0x7f8d9d8fd120]
[0m[38;2;255;000;255m/usr/lib/x86_64-linux-gnu/libGLX_mesa.so.0(+0x44e5b) [0x7f8d9d8fee5b]
[0m[38;2;255;000;255m/usr/lib/x86_64-linux-gnu/libGLX_mesa.so.0(+0x37f68) [0x7f8d9d8f1f68]
[0m[38;2;255;000;255m/usr/lib/x86_64-linux-gnu/libGLX_mesa.so.0(+0x335d4) [0x7f8d9d8ed5d4]
[0m[38;2;255;000;255m/usr/lib/x86_64-linux-gnu/libGLX_mesa.so.0(+0x33b0d) [0x7f8d9d8edb0d]
[0m[38;2;255;000;255m/home/xuyinda/.local/lib/python3.6/site-packages/taichi/core/../lib/taichi_core.so: _glfwInitGLX
[0m[38;2;255;000;255m/home/xuyinda/.local/lib/python3.6/site-packages/taichi/core/../lib/taichi_core.so: _glfwPlatformCreateWindow
[0m[38;2;255;000;255m/home/xuyinda/.local/lib/python3.6/site-packages/taichi/core/../lib/taichi_core.so: glfwCreateWindow
[0m[38;2;255;000;255m/home/xuyinda/.local/lib/python3.6/site-packages/taichi/core/../lib/taichi_core.so: taichi::lang::opengl::initialize_opengl(bool)
[0m[38;2;255;000;255m/home/xuyinda/.local/lib/python3.6/site-packages/taichi/core/../lib/taichi_core.so: taichi::lang::opengl::is_opengl_api_available()
[0m[38;2;255;000;255m/home/xuyinda/.local/lib/python3.6/site-packages/taichi/core/../lib/taichi_core.so(+0x763e47) [0x7f8d9fa54e47]
[0m[38;2;255;000;255m/home/xuyinda/.local/lib/python3.6/site-packages/taichi/core/../lib/taichi_core.so(+0x655bef) [0x7f8d9f946bef]
[0m[38;2;255;000;255m/usr/bin/python3: ) [0x50a635]
[0m[38;2;255;000;255m/usr/bin/python3: _PyEval_EvalFrameDefault
[0m[38;2;255;000;255m/usr/bin/python3: ) [0x509758]
[0m[38;2;255;000;255m/usr/bin/python3: ) [0x50a48d]
[0m[38;2;255;000;255m/usr/bin/python3: _PyEval_EvalFrameDefault
[0m[38;2;255;000;255m/usr/bin/python3: ) [0x507d64]
[0m[38;2;255;000;255m/usr/bin/python3: ) [0x5161e9]
[0m[38;2;255;000;255m/usr/bin/python3: PyCFunction_Call
[0m[38;2;255;000;255m/usr/bin/python3: _PyEval_EvalFrameDefault
[0m[38;2;255;000;255m/usr/bin/python3: ) [0x507d64]
[0m[38;2;255;000;255m/usr/bin/python3: ) [0x509a90]
[0m[38;2;255;000;255m/usr/bin/python3: ) [0x50a48d]
[0m[38;2;255;000;255m/usr/bin/python3: _PyEval_EvalFrameDefault
[0m[38;2;255;000;255m/usr/bin/python3: ) [0x509758]
[0m[38;2;255;000;255m/usr/bin/python3: ) [0x50a48d]
[0m[38;2;255;000;255m/usr/bin/python3: _PyEval_EvalFrameDefault
[0m[38;2;255;000;255m/usr/bin/python3: ) [0x509758]
[0m[38;2;255;000;255m/usr/bin/python3: ) [0x50a48d]
[0m[38;2;255;000;255m/usr/bin/python3: _PyEval_EvalFrameDefault
[0m[38;2;255;000;255m/usr/bin/python3: ) [0x509758]
[0m[38;2;255;000;255m/usr/bin/python3: ) [0x50a48d]
[0m[38;2;255;000;255m/usr/bin/python3: _PyEval_EvalFrameDefault
[0m[38;2;255;000;255m/usr/bin/python3: _PyFunction_FastCallDict
[0m[38;2;255;000;255m/usr/bin/python3: _PyObject_FastCallDict
[0m[38;2;255;000;255m/usr/bin/python3: _PyObject_CallMethodIdObjArgs
[0m[38;2;255;000;255m/usr/bin/python3: PyImport_ImportModuleLevelObject
[0m[38;2;255;000;255m/usr/bin/python3: _PyEval_EvalFrameDefault
[0m[38;2;255;000;255m/usr/bin/python3: ) [0x507d64]
[0m[38;2;255;000;255m/usr/bin/python3: ) [0x5161e9]
[0m[38;2;255;000;255m/usr/bin/python3: PyCFunction_Call
[0m[38;2;255;000;255m/usr/bin/python3: _PyEval_EvalFrameDefault
[0m[38;2;255;000;255m/usr/bin/python3: ) [0x507d64]
[0m[38;2;255;000;255m/usr/bin/python3: ) [0x509a90]
[0m[38;2;255;000;255m/usr/bin/python3: ) [0x50a48d]
[0m[38;2;255;000;255m/usr/bin/python3: _PyEval_EvalFrameDefault
[0m[38;2;255;000;255m/usr/bin/python3: ) [0x509758]
[0m[38;2;255;000;255m/usr/bin/python3: ) [0x50a48d]
[0m[38;2;255;000;255m/usr/bin/python3: _PyEval_EvalFrameDefault
[0m[38;2;255;000;255m/usr/bin/python3: ) [0x509758]
[0m[38;2;255;000;255m/usr/bin/python3: ) [0x50a48d]
[0m[38;2;255;000;255m/usr/bin/python3: _PyEval_EvalFrameDefault
[0m[38;2;255;000;255m/usr/bin/python3: ) [0x509758]
[0m[38;2;255;000;255m/usr/bin/python3: ) [0x50a48d]
[0m[38;2;255;000;255m/usr/bin/python3: _PyEval_EvalFrameDefault
[0m[38;2;255;000;255m/usr/bin/python3: _PyFunction_FastCallDict
[0m[38;2;255;000;255m/usr/bin/python3: _PyObject_FastCallDict
[0m[38;2;255;000;255m/usr/bin/python3: _PyObject_CallMethodIdObjArgs
[0m[38;2;255;000;255m/usr/bin/python3: PyImport_ImportModuleLevelObject
[0m[38;2;255;000;255m/usr/bin/python3: _PyEval_EvalFrameDefault
[0m[38;2;255;000;255m/usr/bin/python3: ) [0x507d64]
[0m[38;2;255;000;255m/usr/bin/python3: ) [0x5161e9]
[0m[38;2;255;000;255m/usr/bin/python3: PyCFunction_Call
[0m[38;2;255;000;255m/usr/bin/python3: _PyEval_EvalFrameDefault
[0m[38;2;255;000;255m/usr/bin/python3: ) [0x507d64]
[0m[38;2;255;000;255m/usr/bin/python3: ) [0x509a90]
[0m[38;2;255;000;255m/usr/bin/python3: ) [0x50a48d]
[0m[38;2;255;000;255m/usr/bin/python3: _PyEval_EvalFrameDefault
[0m[38;2;255;000;255m/usr/bin/python3: ) [0x509758]
[0m[38;2;255;000;255m/usr/bin/python3: ) [0x50a48d]
[0m[38;2;255;000;255m/usr/bin/python3: _PyEval_EvalFrameDefault
[0m[38;2;255;000;255m/usr/bin/python3: ) [0x509758]
[0m[38;2;255;000;255m/usr/bin/python3: ) [0x50a48d]
[0m[38;2;255;000;255m/usr/bin/python3: _PyEval_EvalFrameDefault
[0m[38;2;255;000;255m/usr/bin/python3: ) [0x509758]
[0m[38;2;255;000;255m/usr/bin/python3: ) [0x50a48d]
[0m[38;2;255;000;255m/usr/bin/python3: _PyEval_EvalFrameDefault
[0m[38;2;255;000;255m/usr/bin/python3: _PyFunction_FastCallDict
[0m[38;2;255;000;255m/usr/bin/python3: _PyObject_FastCallDict
[0m[38;2;255;000;255m/usr/bin/python3: _PyObject_CallMethodIdObjArgs
[0m[38;2;255;000;255m/usr/bin/python3: PyImport_ImportModuleLevelObject
[0m[38;2;255;000;255m/usr/bin/python3: ) [0x5141c4]
[0m[38;2;255;000;255m/usr/bin/python3: PyCFunction_Call
[0m[38;2;255;000;255m/usr/bin/python3: _PyEval_EvalFrameDefault
[0m[38;2;255;000;255m/usr/bin/python3: ) [0x507d64]
[0m[38;2;255;000;255m/usr/bin/python3: ) [0x509a90]
[0m[38;2;255;000;255m/usr/bin/python3: ) [0x50a48d]
[0m[38;2;255;000;255m/usr/bin/python3: _PyEval_EvalFrameDefault
[0m[38;2;255;000;255m/usr/bin/python3: ) [0x509758]
[0m[38;2;255;000;255m/usr/bin/python3: ) [0x50a48d]
[0m[38;2;255;000;255m/usr/bin/python3: _PyEval_EvalFrameDefault
[0m[38;2;255;000;255m/usr/bin/python3: _PyFunction_FastCallDict
[0m[38;2;255;000;255m/usr/bin/python3: _PyObject_FastCallDict
[0m[38;2;255;000;255m/usr/bin/python3: _PyObject_CallMethodIdObjArgs
[0m[38;2;255;000;255m/usr/bin/python3: PyImport_ImportModuleLevelObject
[0m[38;2;255;000;255m/usr/bin/python3: _PyEval_EvalFrameDefault
[0m[38;2;255;000;255m/usr/bin/python3: ) [0x507d64]
[0m[38;2;255;000;255m/usr/bin/python3: PyEval_EvalCode
[0m[38;2;255;000;255m/usr/bin/python3: ) [0x634c82]
[0m[38;2;255;000;255m/usr/bin/python3: PyRun_FileExFlags
[0m[38;2;255;000;255m/usr/bin/python3: PyRun_SimpleFileExFlags
[0m[38;2;255;000;255m/usr/bin/python3: Py_Main
[0m[38;2;255;000;255m/usr/bin/python3: main
[0m[38;2;255;000;255m/lib/x86_64-linux-gnu/libc.so.6: __libc_start_main
[0m[38;2;255;000;255m/usr/bin/python3: _start
[0m
```

**Comment by archibate** [comment]
Thank for reporting this, please see https://taichi.readthedocs.io/en/stable/install.html#opengl-issues.
tldr: add `export TI_ENABLE_OPENGL=0` to your `~/.bashrc`.

**Comment by MARMOTatZJU** [comment]
@archibate It works. Thanks a lot!

**Comment by archibate** [comment]
Great! Wish you enjoy Taichi and GAMES 201!
